src/push.rs

```diff
@@ -60,6 +65,18 @@ pub fn push_metrics<S: BuildHasher>(
     push(job, grouping, url, mfs, "PUT", basic_auth)
 }
 
+/// Functions just like `push_metrics`, except the metrics are pushed
+/// asynchronously.
+pub async fn push_metrics_async<S: BuildHasher>(
```

Comment:
For the sake of consistency we should also offer the remaining `pub` functions in an asynchronous fashion, e.g. `push_async_collector`.

Reply:
Agreed, done!